### PR TITLE
assaneRed_update_interfaces

### PR DESCRIPTION
**Pull request by assaneRed**
add TriggerFullActivityDump() and GetFullActivityDump() in interfaces

### CODE REVIEW

File: src/harness/fake_sas.py [hunk 1]
```diff
@@ -247,6 +247,9 @@ def TriggerPpaCreation(self, request, ssl_cert=None, ssl_key=None):
     return 'zone/ppa/fake_sas/%s/%s' % (request['palIds'][0]['palId'],
                                         uuid.uuid4().hex)
 
+  def GetFullActivityDump(self):
```
Comment (sushantparkhi) [inline src/harness/fake_sas.py]: I think you need to move this to class FakeSas and instead implement TriggerFullActivityDump here.

File: src/harness/sas.py [hunk 2]
```diff
@@ -142,6 +142,9 @@ def GetSasImplementationRecord(self, request, ssl_cert=None, ssl_key=None):
   def GetEscSensorRecord(self, request, ssl_cert=None, ssl_key=None):
     return self._SasRequest('esc_sensor', request, ssl_cert, ssl_key)
 
+  def GetFullActivityDump(self, ssl_cert=None, ssl_key=None):
+    return self._SasRequest('dump', None, ssl_cert, ssl_key)
```
Comment (sushantparkhi) [inline src/harness/sas.py]: If you are planning to pass None in the _SasRequest, I think you need to update the definition of the _SasRequest and may be something like `'https://%s'%'/'.join(value for value in [self._base_url, self._sas_version, method_name, request] if value)` in the _RequestGet function

File: src/harness/sas_interface.py [hunk 3]
```diff
@@ -187,6 +187,13 @@ def GetEscSensorRecord(self, request, ssl_cert=None, ssl_key=None):
     """
     pass
 
+  @abc.abstractmethod
+  def GetFullActivityDump(self, ssl_cert=None, ssl_key=None):
+    """SAS-SAS Full Activity Dump  interface Requests a Pull Command to get Full Activity Dump Message
```
Comment (sushantparkhi) [inline src/harness/sas_interface.py]: May be a removal of multiple spaces from comments?

File: src/harness/sas.py [hunk 4]
```diff
@@ -267,6 +270,11 @@ def TriggerPpaCreation(self, request):
                         self._GetDefaultAdminSSLCertPath(),
                         self._GetDefaultAdminSSLKeyPath())
 
+  def TriggerFullActivityDump(self):
+    _RequestPost('https://%s/admin/trigger/create_full_activity_dump' % self._base_url, None,
```
Comment (sushantparkhi) [inline src/harness/sas.py]: May be break this line to consolidate it to 80 characters?

**Comment by assaneRed** [comment]
Hi sushantparkhi,
I fixed bug based on your comments 
Can you check please? and let me know if there any other comments

Thanks

**Comment by assaneRed** [comment]
Hi sushantparkhi,
I fixed bug based on your comments
Can you check please? and let me know if there any other comments

Best regards,

File: src/harness/fake_sas.py [hunk 5]
```diff
@@ -192,6 +192,9 @@ def GetEscSensorRecord(self, request, ssl_cert=None, ssl_key=None):
       # Return Empty if invalid Id
       return {}
 
+  def GetFullActivityDump(self):
```
Comment (sushantparkhi) [inline src/harness/fake_sas.py]: Thanks for the changes! I think the function signature is not matching with an interface `def GetFullActivityDump(self, ssl_cert=None, ssl_key=None):`

Also, you need to add this to `do_GET()` in fake_sas.py?

File: src/harness/fake_sas.py [hunk 6]
```diff
@@ -192,6 +192,9 @@ def GetEscSensorRecord(self, request, ssl_cert=None, ssl_key=None):
...
+    return {'files':[{'url':None, 'checksum':None, 'size':0, 'version':None, 'recordType':None}]}
```
Comment (sushantparkhi) [inline src/harness/fake_sas.py]: An addition of whitespace after each key and breaking this line to have 80 characters in each line?

**Comment by assaneRed** [comment]
Hi Sushant,
Thanks for your comments. 
Changes are done; you can check again if you have time
Best regards

File: src/harness/fake_sas.py [hunk 7]
```diff
@@ -192,6 +192,10 @@ def GetEscSensorRecord(self, request, ssl_cert=None, ssl_key=None):
       # Return Empty if invalid Id
       return {}
 
+  def GetFullActivityDump(self, ssl_cert=None, ssl_key=None):
+    return {'files':[{'url': None, 'checksum': None, 'size': 0, 'version': None,
+           'recordType': None}]}
```
Comment (sushantparkhi) [inline src/harness/fake_sas.py]: This line looks like it is under-indented?

File: src/harness/sas_interface.py (hunk 3 above)
Comment (sushantparkhi) [inline src/harness/sas_interface.py]: I think this line contains more than 80 characters, may be breaking this line?

**Comment by kate-harrison** [comment]
Since this has been approved, please fix the merge conflicts and merge this into master.

**Pull request closed without merging**